spec/integration/system_pager_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+RSpec.describe TTY::Pager::SystemPager do
+  around :each do |example|
+    Dir.mktmpdir do |dir|
+      FileUtils.cd(dir) do
+        example.run
+      end
+    end
+  end
+
+  it "paginates some text" do
+    output_path = Pathname.new('output.txt')
+
+    described_class.page("Some text", command: "cat > #{output_path}")
+    expect(output_path.read).to eq "Some text"
+  end
+
+  it "allows pagination to happen asynchronously" do
+    output_path = Pathname.new('output.txt')
+    File.open('external_pager.rb', 'w') do |script|
+      script.write(<<~RUBY)
+        File.open('output.txt', 'w') do |output|
+          ARGF.each { |line| output.puts(line) }
+        end
+      RUBY
+    end
+
+    described_class.page(command: "ruby external_pager.rb") do |pager|
+      pager.puts("one")
+      pager.write("two\n")
+      pager.try_write("three\n")
+    end
+
+    expect(output_path.read).to eq "one\ntwo\nthree\n"
+  end
+
+  it "stops paginating once external tool is closed" do
+    output_path = Pathname.new('output.txt')
+    File.open('external_pager.rb', 'w') do |script|
+      script.write(<<~RUBY)
+        ARGF.each_with_index do |line, index|
+          File.open('output.txt', 'a') do |output|
+            output.puts(line)
+            output.flush
+          end
+        end
+      RUBY
+    end
+
+    described_class.page(command: "ruby external_pager.rb") do |pager|
+      pager.puts("one")
+      pager.puts("two")
+
+      # wait for script to finish writing
+      sleep 0.5
+
+      pid = pager.instance_variable_get('@pager_io').instance_variable_get('@pid')
+      Process.kill('TERM', pid)
+      Process.waitpid(pid)
```

Comment:
This raises on Windows with an invalid argument for `TERM`.  Changing to `Process.kill("KILL", pid)` works for me in Windows VM.